qiskit/primitives/containers/bindings_array.py

```diff
@@ -364,9 +368,12 @@ def examine_array(*possible_shapes):
             # the last dimension _has_ to be over parameters
             examine_array(val.shape[:-1])
         elif val.shape and val.shape[-1] == 1:
-            # this case is a convention, and separated from the previous case for clarity:
-            # if the last axis is 1-d, make an assumption that it is for our 1 parameter
-            examine_array(val.shape[:-1])
+            if val.ndim == 1:
```

Comment:
The only case where it is possible is if `val.shape == (1,)`. If so then I prefer `val.shape == (1,)` over `val.ndim==1` since it's more readable and explicit.